src/components/v2/V2Project/V2FundingCycleSection/FundingCycleHistory.tsx

```diff
@@ -22,6 +22,94 @@ import useUsedDistributionLimit from 'hooks/v2/contractReader/UsedDistributionLi
 import { V2UserContext } from 'contexts/v2/userContext'
 import { formatDiscountRate, MAX_DISTRIBUTION_LIMIT } from 'utils/v2/math'
 
+// Fill in gaps between first funding cycle of each configuration:
+//     - derives starts from duration and start time of the first FC of that configuration
+//     - derives weights from discount rate and weight of the first FC of the configuration
+//     - derives number by incrementing
+//     - everything else the same as the first FC of the configuration
+const deriveFundingCyclesBetweenEachConfiguration = ({
```

Comment:
Move this Fn outside the component as it doesn't need to be there. Helps with maintainability.